GUIDELINES.md

```diff
@@ -1,3 +1,694 @@
 # CA Technologies C# Style Guide
 
-## Please add content
\ No newline at end of file
+This document describes the C# coding style of the Xamarin Mobile SDK team. This guideline is recommended to comply with all C# implementations of our products.
+
+## Review Microsoft Official Coding Guideline
+
+Beyond the guidelines defined in this document, we also recommend reviewing the [Microsoft C# Guide](https://docs.microsoft.com/en-us/dotnet/csharp/index). 
+
+### Mobile SDK
+
+For more information, see [developer website](http://mas.ca.com).
+
+The Xamarin Mobile SDK consists of multiple functional products separated into individual frameworks.
+
+All of frameworks are recommended to comply with the coding guideline.
+
+## Table of Contents
+
+* [Documentation](#documentation)
+* [Brace Style](#brace-style)
+* [New Line](#new-line)
+* [Wrapping Lines](#wrapping-lines)
+  * [Long Argument Lists](#long-argument-lists) 
+  * [Strings](#strings)
+* [Lambdas](#lambdas)
+* [Spacing](#spacing)
+  * [Vertical Spacing](#vertical-spacing)
+  * [Blocks](#blocks)
+  * [Properties](#properties)
+  * [Other Cases](#other-cases)
+* [Enums](#enums)
+* [#region directive](#region-directive)
+* [using directive](#using-directive)
+* [Nomenclature](#nomenclature)
+* [Comments](#comments)
+  * [In-line comment](#in-line-comment)
+  * [Multiline comment](#multiline-comment)
+* [File Structure](#file-structure)
+* [Copyright Statement](#copyright-statements)
+* [Error Handling](#error-handling)
+
+
+## Documentation
+
+All code **should be properly documented** in `.cs` files. Code documentation should adhere to Microsoft Doc's format.
+
+Microsoft provides a documentation generation system based on XML comments. These comments are formally single line C♯ comments (indicated by triple slashes) containing XML tags. All comments must be directly before the code block to which the comments refer, for example:
```

Comment:
All comments must be placed directly before the code block to which the comments refer, for example: